### PR TITLE
Add cache for Docker images in CI

### PR DESCRIPTION
**Pull request by iblancasa**
Signed-off-by: Israel Blancas <iblancas@redhat.com>

## Which problem is this PR solving?
- #1796

## Short description of the changes
- Use the `actions/cache` GitHub Actions cache to cache the build
- Enable the usage of build kit
- Add an extra variable to `docker build` command to indicate `docker build` where the cache is located

### CODE REVIEW

File: Dockerfile.asserts [hunk 1]
```diff
@@ -1,26 +1,28 @@
...
-COPY go.mod /go/src/github.com/jaegertracing/jaeger-operator/go.mod
-COPY go.sum /go/src/github.com/jaegertracing/jaeger-operator/go.sum
+COPY go.mod go.mod
+COPY go.sum go.sum
 RUN go mod download
 
-COPY . /go/src/github.com/jaegertracing/jaeger-operator/
+COPY tests tests
+
+ENV CGO_ENABLED=0
```
Comment (frzifus) [inline Dockerfile.asserts:15]: As far as i know thats a default value since go 1.16.

Comment (iblancasa) [reply Dockerfile.asserts:15]: I don't find anything in the documentation. Could you point to a reference? [I specially checked the Go 1.16 release notes](https://go.dev/doc/go1.16)

Thanks!

Comment (frzifus) [reply Dockerfile.asserts:15]: oh you are right. Mixed it up with [GO111MODULE](https://maelvls.dev/go111module-everywhere/) sorry.

**Review by rubenvp8510 — DISMISSED** [review]

**Comment by rubenvp8510** [comment]
Overall looks good, please solve the conflict.

**Review by kevinearls — APPROVED** [review]

**Review by rubenvp8510 — APPROVED** [review]

**Pull request merged**